onedal/datatypes/utils/common.cpp

```diff
@@ -0,0 +1,41 @@
+/*******************************************************************************
+* Copyright 2024 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include <cstdint>
+
+namespace oneapi::dal::python {
+
+union endian_checker {
+    std::uint64_t blob;
+    std::uint8_t arr[8];
+};
+
+bool is_little_endian_impl() {
+    constexpr std::uint64_t one = 0xfful;
+    constexpr endian_checker checker{ one };
+    return static_cast<bool>(checker.arr[0]);
```

Comment:
Is the return of this static cast done at runtime or compile time? the constexpr before I am certain of, but the actual check of the array and conversion seem to be done at runtime

Reply:
It was bad designed.
Please follow discussion here
https://github.com/intel/scikit-learn-intelex/pull/2045/files#r1807940521